Lib/traceback.py

```diff
@@ -212,11 +212,16 @@ def extract_stack(f=None, limit=None):
 def clear_frames(tb):
     "Clear all references to local variables in the frames of a traceback."
     while tb is not None:
-        try:
-            tb.tb_frame.clear()
-        except RuntimeError:
-            # Ignore the exception raised if the frame is still executing.
-            pass
+        frame = tb.tb_frame
+        while True:
```

Comment:
This is making `clear_frames()` a potentially O(n**2) operation. In practice, you don't need this, as `tb.tb_next.tb_frame.f_back` will be the same as `tb.tb_frame` (and so on). So the loops needn't be nested.

Reply:
Or you can keep the nesting but mark frames as seen as you iterate on them, which will have the same effect while being more general (in case someone changes the traceback chain).

Reply:
> This is making clear_frames() a potentially O(n**2) operation.

Sorry, I don't know well how traceback objects are created. I didn't even know that you can have two tracebacks objects linked together.

I added a "seen = set()" to prevent iterating twice on the same frame chain. Does it solve your O(n**2) issue?

I also used my example attached to the bpo to enhance the existing unit test.

Reply:
You can also use `while frame is not None` here instead of having a separate test at the end of the loop.